pkg/agent/baker.go

```diff
@@ -264,8 +264,8 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 			return profile.CustomLinuxOSConfig.TransparentHugePageDefrag
 		},
 		"ShouldConfigSwapFile": func() bool {
-			// only configure swap file when FailSwapOn is true and SwapFileSizeMB is valid
-			return profile.CustomKubeletConfig != nil && profile.CustomKubeletConfig.FailSwapOn != nil && *profile.CustomKubeletConfig.FailSwapOn &&
+			// only configure swap file when FailSwapOn is false and SwapFileSizeMB is valid
```

Comment:
what is the default behavior of this FailSwapOn ? true or false?

Reply:
Default is true. It makes kubelet fail to start if node has swap. So we have to explicitly turn it off to use swap.
https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/#:~:text=--fail-swap-on

Reply:
ok. so we should have validation in RP to error out if failswapon is true and swapFileSizeMB is not nil

Reply:
Yes.. there's actually quite a lot of validation we need to do